File: Backend/src/Modules/tour/Dao/tour.dto.ts
import { Prisma } from '@prisma/client';
import { IsArray, IsEnum, IsNotEmpty, IsOptional } from 'class-validator';
import locations from '../interfaces/location.interface';

class TourDto {
  id: string;

  name: string;

  @IsNotEmpty()
  duration: number;

  @IsNotEmpty()
  maxGroupSize: number;

  @IsEnum(['easy', 'medium', 'hard'], { each: true })
  difficulty: string;

  @IsOptional()
  ratingsAverage: Prisma.Decimal;

  @IsOptional()
  ratingsQuantity: number;

  @IsNotEmpty()
  price: number;

  @IsOptional()
  priceDiscount: number | null;

  summary: string;

  startDates: Date[];

  description: string;
  coverImage: string | null;
  startLocation: locations;

  @IsArray()
  locations: locations[];
}
export default TourDto;

File: app/src/main/java/com/tourguide/data/dto/TourDto.kt
package com.tourguide.data.dto

import com.tourguide.domain.model.Location
import com.tourguide.domain.model.Tour
import com.tourguide.domain.model.User
import java.util.*

data class TourDto(
    val difficulty: String,
    val duration: Int,
    val durationWeeks: Double,
    val guides: List<User>,
    val id: String,
    val imageCover: String,
    val images: List<String>,
    val locations: List<Location>,
    val maxGroupSize: Int,
    val description: String,
    val name: String,
    val price: Double,
    val rating: Double,
    val ratingsAverage: Double,
    val ratingsQuantity: Int,
    val secretTour: Boolean,
    val slug: String,
    val startDates: List<Date>,
    val startLocation: Location,
    val summary: String,
)

fun TourDto.toTour(): Tour {
    return Tour(
         id = id,
         slug = slug,
         name = name,
         duration = duration,
         maxGroupSize = maxGroupSize,
         difficulty = difficulty,
         averageRating = ratingsAverage,
         price = price,
         summary = summary,
         description = description,
         thumbnail = imageCover,
         images = images,
         startDates = startDates,
         startLocation = startLocation,
         locations = locations,
         guides = guides,
    )
}